prime-router/src/main/kotlin/azure/GetSubmissions.kt

```diff
@@ -0,0 +1,85 @@
+package gov.cdc.prime.router.azure
+
+import com.microsoft.azure.functions.HttpMethod
+import com.microsoft.azure.functions.HttpRequestMessage
+import com.microsoft.azure.functions.HttpResponseMessage
+import com.microsoft.azure.functions.annotation.AuthorizationLevel
+import com.microsoft.azure.functions.annotation.FunctionName
+import com.microsoft.azure.functions.annotation.HttpTrigger
+import gov.cdc.prime.router.tokens.OktaAuthentication
+import org.apache.logging.log4j.kotlin.logger
+import java.time.OffsetDateTime
+
+/**
+ * Submissions API
+ * Returns a list of Actions from `public.action`.
+ */
+
+class GetSubmissions(
+    submissionsFacade: SubmissionsFacade = SubmissionsFacade.common,
+    oktaAuthentication: OktaAuthentication = OktaAuthentication(PrincipalLevel.SYSTEM_ADMIN)
+) {
+    private val facade = submissionsFacade
+    private val oktaAuthentication = oktaAuthentication
+
+    /**
+     * An Azure Function that is triggered at the `/api/submissions/` endpoint
+     *
+     * @param qLimit a user supplied page size limit overwriting the default value.
+     * @param limit a default value for the number of results to display per page.
+     * @return a list of submission history results.
+     */
+    @FunctionName("getSubmissions")
+    fun run(
+        @HttpTrigger(
+            name = "getSubmissions",
+            methods = [HttpMethod.GET],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "submissions"
+        ) request: HttpRequestMessage<String?>,
+    ): HttpResponseMessage {
+        val qOrder = request.queryParameters.getOrDefault("order", "DESC")
+        val qCursor = request.queryParameters.getOrDefault("cursor", "${OffsetDateTime.now()}")
+        val qLimit = request.queryParameters.getOrDefault("limit", "10")
+
+        if (isPositiveInteger(qLimit)) {
+            val order = qOrder
+            val cursor = OffsetDateTime.parse(qCursor)
+            val limit = qLimit.toInt()
+            return getList(request, order, cursor, limit)
+        }
+
+        return HttpUtilities.bad(request, "Limit must be a positive integer.")
+    }
+
+    /**
+     * Return true if [s] is a positive integer.
+     * TODO: move to a utility class
+     */
+    private fun isPositiveInteger(s: String): Boolean {
+        return !s.isNullOrEmpty() && s.all { Character.isDigit(it) } && s.toInt() > 0
+    }
+
+    /**
+     * @param request the body content from an HTTP Request to pass into Okta for authentication.
+     * @param limit is an Integer used for setting the number of results per page.
+     * @return data after Okta Authentication
+     */
+    private fun getList(
+        request: HttpRequestMessage<String?>,
+        order: String,
+        cursor: OffsetDateTime,
+        limit: Int
+    ): HttpResponseMessage {
+        return oktaAuthentication.checkAccess(request, "") {
```

Comment:
Move okta auth and pulling things out of the jwt in the top level entrypoint function - no one should be able to access this anonymously.